### ISSUE TITLE
ExecutorService can be used instead of ScheduledExecutorService

### ISSUE THREAD

**Issue by hendrikebbers** [eclipse-paho/paho.mqtt.java#603]
The MqttAsyncClient needs a ScheduledExecutorService param for the constructor but only the functionality of a ExecutorService is needed. Other frameworks works more often with ExecutorService and provides good functionality to create such a service. Based on all that points ExecutorService should be used instead.

**Comment by hendrikebbers** [comment]
@dobermai just started to have a deeper look ;)

**Comment by icraggs** [comment]
Hello.  What is the benefit of using ExecutorService rather than ScheduledExecutorService?

**Comment by hendrikebbers** [comment]
less functionality in the interface:
ScheduledExecutorService extends ExecutorService
No functionality of ScheduledExecutorService is used. Only the functionality of ExecutorService was used internally.

**Comment by icraggs** [comment]
Fair.  I don't know if you are using this functionality, but the library attempts to shut down the supplied executor service when the client object is closed.  This doesn't make sense to me but I could be missing something.  Is there a situation where this does make sense?

**Comment by hendrikebbers** [comment]
From my point of view this is a big problem. Would be something I will try to work one once this issue is solved. Step by step ;)

**Comment by icraggs** [comment]
I've made changes to the develop branch to a) only use an executor service when provided, otherwise to use 'normal' threads and b) not to shutdown a supplied executor service.  These changes exist now for the MQTT V3 client, I'm just making the equivalent for MQTT V5.

a) solves another issue where threads weren't stopping when not in use when an executor service wasn't supplied.

**Comment by hendrikebbers** [comment]
@icraggs is there a PR for the change? Would love to have a look at the changes

**Comment by icraggs** [comment]
No, there is no PR, I've been committing directly to the develop branch, which is what I usually do on my own repos.

The commits start around: https://github.com/eclipse/paho.mqtt.java/commit/576455ef909a1c81a7fd6c93e5508b7b289e55fc

The result is the latest version of the develop branch.

**Comment by bpair** [comment]
Would you want a scheduled service if you were using Websockets/OAuth2 and refreshing a token? I am just trying to think of possible desired functionality.
